### PR TITLE
Refactor tests - add arrow functions

### PR DESCRIPTION
**Pull request by stojanovic**
Again arrow functions.
However, I changed only one line functions, and I left something like this as it was:
```
app.on('error', function(err){
  err.message.should.equal('boom');
  done();
});
```
Not sure if we are getting anything with this:
```
app.on('error', err => {
  err.message.should.equal('boom');
  done();
});
```
except maybe consistency (using arrow functions for all callbacks) and 4-6 characters less.

### CODE REVIEW

**Comment by tejasmanohar** [comment]
I usually just use all arrow functions for arguments and such to be consistent, but correct- there's no benefit as you mentioned so maybe it's not worth the PR diff.

**Comment by stojanovic** [comment]
There's one more "special case", this doesn't work:
```
ctx.res.end = () => throw new Error('response sent');
```
It needs to be written like this:
```
ctx.res.end = () => {
  throw new Error('response sent');
};
```
So I [left it unchanged](https://github.com/koajs/koa/blob/master/test/application/index.js#L36).

**Comment by stojanovic** [comment]
@tejasmanohar I agree, but that should be a part of a separate PR since it wasn't like that earlier. I can do it but not sure if everyone thinks the same.

**Comment by tejasmanohar** [comment]
@stojanovic Well, previously the prop `.should.be.ok` was directly chained to `)`, now it's on a new line so indentation is needed for it to be readable imo.

**Comment by stojanovic** [comment]
@tejasmanohar moved to the same line. +1 for indentation everywhere for that situation.

File: test/application/onerror.js [hunk 1]
```diff
@@ -9,9 +9,7 @@ describe('app.onerror(err)', function(){
   it('should throw an error if a non-error is given', function(done){
     const app = new Koa();
 
-    (function(){
-      app.onerror('foo');
-    }).should.throw(AssertionError, {message: 'non-error thrown: foo'});
+    (() => app.onerror('foo')).should.throw(AssertionError, {message: 'non-error thrown: foo'});
```
Comment (juliangruber) [inline test/application/onerror.js]: do we care about the 80 chars limit still?

Comment (stojanovic) [inline test/application/onerror.js]: I was trying to keep it under 80 chars, but there's a few longer lines so I just moved this to the same line after a comment.

**Comment by jonathanong** [comment]
meh. the only reason i'd merge this is because future PRs for ES6 refactoring in inevitable.

if we make these sort of changes, can we update the eslint file with relevant rules?

**Comment by stojanovic** [comment]
@jonathanong done.
There's one more rule, but it needs to be added in #551:
```
  prefer-arrow-callback: 2
```
cc. @broucz

**Comment by jonathanong** [comment]
great :+1:

**Pull request closed without merging**